captum/metrics/_core/infidelity.py

```diff
@@ -523,7 +540,25 @@ def _next_infidelity(current_n_perturb_samples: int) -> Tensor:
         metrics_sum = _divide_and_aggregate_metrics(
             cast(Tuple[Tensor, ...], inputs),
             n_perturb_samples,
-            _next_infidelity,
+            _next_infidelity_tensors,
+            agg_func=_concat_infidelity_tensors,
             max_examples_per_batch=max_examples_per_batch,
         )
-    return metrics_sum * 1 / n_perturb_samples
+
+    attr_times_perturb_sums, perturbed_fwd_diffs = metrics_sum
```

Comment:
nit: Can include this above as return of _divide_and_aggregate_metrics ? metrics_sum isn't really applicable anymore